tests_integration/nativeapp/test_bundle.py

```diff
@@ -29,56 +29,76 @@
 TEST_ENV = generate_user_env(USER_NAME)
 
 
-@pytest.fixture
-def template_setup(runner, temporary_working_directory):
-    project_name = "myapp"
-    result = runner.invoke_json(
-        ["app", "init", project_name],  # Uses default template
-        env=TEST_ENV,
-    )
-    assert result.exit_code == 0
+@pytest.fixture(scope="function", params=["v1", "v2"])
+@enable_definition_v2_feature_flag
+def template_setup(runner, project_directory, request):
+    definition_version = request.param
+    with project_directory(f"napp_init_{definition_version}") as project_root:
+        # Vanilla bundle on the unmodified template
+        result = runner.invoke_json(
+            ["app", "bundle"],
+            env=TEST_ENV,
+        )
+        assert result.exit_code == 0
+
+        # The newly created deploy_root is explicitly deleted here, as bundle should take care of it.
 
-    # Vanilla bundle on the unmodified template
-    result = runner.invoke_json(
-        ["app", "bundle", "--project", project_name],
-        env=TEST_ENV,
-    )
-    assert result.exit_code == 0
-    # The newly created deploy_root is explicitly deleted here, as bundle should take care of it.
+        deploy_root = Path(project_root, "output", "deploy")
+        assert Path(deploy_root, "manifest.yml").is_file()
+        assert Path(deploy_root, "setup_script.sql").is_file()
+        assert Path(deploy_root, "README.md").is_file()
 
-    project_root = Path(temporary_working_directory, project_name)
-    deploy_root = Path(project_root, "output", "deploy")
-    assert Path(deploy_root, "manifest.yml").is_file()
-    assert Path(deploy_root, "setup_script.sql").is_file()
-    assert Path(deploy_root, "README.md").is_file()
+        return project_root, runner, definition_version
 
-    return project_root, runner
+
+def override_snowflake_yml_artifacts(
+    definition_version, artifacts_section, deploy_root=str(Path("output", "deploy"))
+):
+    with open("snowflake.yml", "w") as f:
+        if definition_version == "v2":
+            file_content = f"""
+                definition_version: 2
+                entities:
+                  pkg:
+                    type: application package
+                    name: myapp_pkg_<% ctx.env.USER %>
+                    artifacts:
+                {indent(artifacts_section, "                    ")}
```

Comment:
```suggestion
                {indent(artifacts_section, " " * 20)}
```

Reply:
this seems really brittle though, how about using `yaml.dump` to dump a `dict`(and pass `artifacts_section` as a `list`) instead of dealing with raw strings?